easybuild/base/generaloption.py

```diff
@@ -232,7 +235,9 @@ def store_or(option, opt_str, value, parser, *args, **kwargs):  # pylint: disabl
                 """Callback for supporting options with optional values."""
                 # see http://stackoverflow.com/questions/1229146/parsing-empty-options-in-python
                 # ugly code, optparse is crap
-                if parser.rargs and not parser.rargs[0].startswith('-'):
+                if option.store_or == self.STORE_OR_FALSE and opt_str.startswith("--%s-" % self.DISABLE):
```

Comment:
2nd part of this condition feels very hard-codey...

Is that also how we can for disabling of regular boolean options?

Reply:
Yes, that is basically copied verbatim from https://github.com/easybuilders/easybuild-framework/blob/a42e25be0300a263747c6742c06197b7fcdcddf6/easybuild/base/generaloption.py#L294-L297